opal/corelib/string/unpack.rb

```diff
@@ -686,6 +686,33 @@ class String
       'p': null
     }
 
+    function toArray(callback) {
+      return function(data) {
+        return [callback(data)];
+      }
+    }
+
+    var optimized = {
+      'C*': identityFunction,
+      'c*': toNByteSigned(1, identityFunction),
+      'A*': wrapIntoArray(joinChars(bytesToAsciiChars(filterTrailingZerosAndSpaces(identityFunction)))),
+      'a*': wrapIntoArray(joinChars(bytesToAsciiChars(identityFunction))),
+      'M*': toArray(qpdecode(joinChars(bytesToAsciiChars(identityFunction)))),
+      'm*': toArray(base64Decode(joinChars(bytesToAsciiChars(identityFunction)))),
+      'S*': mapChunksToWords(chunkBy(2, identityFunction)),
+      's*': toNByteSigned(2, mapChunksToWords(chunkBy(2, identityFunction))),
+      'L*': mapChunksToWords(chunkBy(4, identityFunction)),
+      'l*': toNByteSigned(4, mapChunksToWords(chunkBy(4, identityFunction))),
+      'Q*': mapChunksToWords(chunkBy(8, identityFunction)),
+      'q*': toNByteSigned(8, mapChunksToWords(chunkBy(8, identityFunction))),
+      'S>*': mapChunksToWords(invertChunks(chunkBy(2, identityFunction))),
```

Comment:
Not sure about this notation `S>*` ?

Reply:
Why?
``` ruby
('a'..'z').to_a.join.unpack("S>*")
 => [24930, 25444, 25958, 26472, 26986, 27500, 28014, 28528, 29042, 29556, 30070, 30584, 31098]

Reply:
I wasn't sure that `S>*` was a valid format